misc/pytorch_toolkit/breast_ultrasound_simulation/README.md

```diff
@@ -0,0 +1,137 @@
+We further derive the relationship between the frequency of the transducer and the resolution of the ultrasound image to simulate and form images at frequencies other than at which the model is trained. This is achieved by interpolation between dilated kernels in a multi-linear separable CNN used to control the resolution of the generated image allowing for zero-shot adaptation. Using the multi-linear separable design, we can simulate entire 3D volumes by training the model only on 2D ultrasound images. 
+Given a  CNN trained on a  2D ultrasound dataset acquired using a transducer of a particular frequency, we can simulate ultrasound images and 3D volumes corresponding to various transducer frequencies without the need for additional training data in 3D and at a different frequency. We also demonstrate the ability to simulate ultrasound images of breast tumour using a model trained on the IVUS dataset. We have performed an extensive validation of our approach using both real and simulated ultrasound images. Using regression analysis, we have concluded that there exists a linear relationship between the proposed kernel interpolation factor and the transducer frequency, thereby laying the foundation of a multi-frequency full 3D ultrasound volume simulator.
+
+
+
+## Proposed Approach
+
+<p align="center"><img src="./media/graphic.jpg" alt="drawing" width="500"/></p>
+
+## Results
+
+![](./media/cart.jpg)
+
+## Setup
+### Prerequisites
+ * python 3.7.3
+ * torch                  1.5.0
+ * torchvision            0.6.0
+ * numpy                  1.19.2
+ * opencv-python          4.5.1.48
+ * Pillow                 7.2.0
+
+```
+sh init_venv.sh
+```
+## Datasets
+The network is trained on [IVUS](http://www.cvc.uab.es/IVUSchallenge2011/dataset.html) dataset. Data augmentation is applied as explain in [1,2,3] and images are processed by [stage 0](https://in.mathworks.com/matlabcentral/fileexchange/34199-pseudo-b-mode-ultrasound-image-simulator). We have directly provided the processed dataset [here]([link](https://drive.google.com/drive/folders/1d4iu2OHxSaORK4mPXqb0Wy2ivXbXZIdI?usp=sharing)). Place the processed data as shown below.
+```
++-- data
+|   +-- stage0
+    |   +-- frame_03_0008_003_240_down.png
+    |   +-- frame_03_0008_003_120_down.png
+                        .
+                        .
+|   +-- real_images
+    |   +-- frame_03_0008_003_240_down.png
+    |   +-- frame_03_0008_003_120_down.png
+                        .
+                        .
+```
+Segmentation masks from [INbreast](http://medicalresearch.inescporto.pt/breastresearch/index.php/Get_INbreast_Database) dataset is used to generate stage 0 simulation results for breast ultrasound. Alternate source for INbreast dataset [link](https://www.kaggle.com/ramanathansp20/inbreast-dataset). Note that, we have not used INbreast dataset to train these models. We demonstrate the ability to simulate breast ultrasound images using a model trained on IVUS dataset.
```

Comment:
The first link (medical research) is dead. No need to store it, you can remove it.